src/main/java/com/sequenceiq/cloudbreak/service/stack/resource/azure/builders/AzureResourceBuilderInit.java

```diff
@@ -34,40 +34,41 @@
         AzureProvisionContextObject azureProvisionContextObject =
-                new AzureProvisionContextObject(stack.getId(), credential.getCommonName(template.getLocation()), azureClient,
-                        getOsImageName(credential, template.getLocation()), userData);
+                new AzureProvisionContextObject(stack.getId(), credential.getCommonName(azureLocation), azureClient,
+                        getOsImageName(credential, azureLocation), userData);
         return azureProvisionContextObject;
     }
 
     @Override
     public AzureDeleteContextObject deleteInit(Stack stack) throws Exception {
         AzureCredential credential = (AzureCredential) stack.getCredential();
-        AzureTemplate template = (AzureTemplate) stack.getTemplate();
+        AzureLocation azureLocation = AzureLocation.valueOf(stack.getRegion());
 
-        AzureClient azureClient =  azureStackUtil.createAzureClient((AzureCredential) stack.getCredential());
+        AzureClient azureClient = azureStackUtil.createAzureClient((AzureCredential) stack.getCredential());
         AzureDeleteContextObject azureDeleteContextObject =
-                new AzureDeleteContextObject(stack.getId(), credential.getCommonName(template.getLocation()), azureClient);
+                new AzureDeleteContextObject(stack.getId(), credential.getCommonName(azureLocation), azureClient);
         return azureDeleteContextObject;
     }
 
     @Override
     public AzureDeleteContextObject decommissionInit(Stack stack, Set<String> decommissionSet) throws Exception {
         AzureCredential credential = (AzureCredential) stack.getCredential();
-        AzureTemplate template = (AzureTemplate) stack.getTemplate();
+        AzureLocation azureLocation = AzureLocation.valueOf(stack.getRegion());
 
-        AzureClient azureClient =  azureStackUtil.createAzureClient((AzureCredential) stack.getCredential());
+        AzureClient azureClient = azureStackUtil.createAzureClient((AzureCredential) stack.getCredential());
         List<Resource> resourceList = new ArrayList<>();
         for (String res : decommissionSet) {
-            resourceList.add(new Resource(ResourceType.AZURE_VIRTUAL_MACHINE, res, stack));
-            resourceList.add(new Resource(ResourceType.AZURE_SERVICE_CERTIFICATE, res, stack));
-            resourceList.add(new Resource(ResourceType.AZURE_CLOUD_SERVICE, res, stack));
+            List<InstanceGroup> instanceGroups = Lists.newArrayList(stack.getInstanceGroups());
+            resourceList.add(new Resource(ResourceType.AZURE_VIRTUAL_MACHINE, res, stack, instanceGroups.get(0).getGroupName()));
+            resourceList.add(new Resource(ResourceType.AZURE_SERVICE_CERTIFICATE, res, stack, instanceGroups.get(0).getGroupName()));
+            resourceList.add(new Resource(ResourceType.AZURE_CLOUD_SERVICE, res, stack, instanceGroups.get(0).getGroupName()));
```

Comment:
why do we build up a new Resource set before decommisioning? These resources are persisted and can be queried. That way it won't be needed to pass a random instanceGroup here again.